### ISSUE TITLE
lessons.md reviewed

### ISSUE THREAD

**Issue by kallewesterling** [DHRI-Curriculum/python#31]
- Look at your workshop on [the website](http://kallewesterling.pythonanywhere.com) to see what your workshop currently looks like, and what looks like it needs to change.
- Think about order and transitions—do the lessons follow a logical pattern and is it working? Do you need to add something somewhere? Are the transitions between the lessons clear?
- Can you add a challenge at the end of the lesson that summarizes what the learner has learned, and activates the knowledge in an interactive way.
- Think about the length of the lessons — are they too long? Too short? In general, each lesson should cover one topic, or one particular skill — or perhaps connect two skills. If you find that you’re covering more than that in one lesson, perhaps they should be split into more than one.
- While reviewing the lessons.md file, think about vocabulary terms that we should add in an easily accessible "dictionary" for our learners.

Feel free to reach out to me and @lmrhody and ask for help, if you need it. Also feel free to involve others in the group!

**Comment by kallewesterling** [comment]
If you want another view of the Markdown and how it is being translated/parsed into the website, you can see [this link](https://kallewesterling.pythonanywhere.com/preview). It is a new tool that I created that will download the live markdown from GitHub and preview the way that the website interprets the markdown. In order to access it, you'll have to be logged in using `admin` as your login and `admin` as your password!

Once you're logged in, you can click "Preview in parser" and it should show you what the live markdown is, and how the website will interpret your markdown.

**Comment by paramajmera** [comment]
Here's my thoughts and suggestions for the python workshop - 

- I mentioned this in my notes on the git workshop, but I would recommend moving the green "get started" button to appear below the "contexts" section in the first page of the workshop.
- I really love the book names that you used as your examples!
- I think that the challenges and solutions are really helpful and well implemented.
- I think that this workshop could be enhanced with a concluding section that helps participants imagine what they might do with python and connect python to other workshops like git, text analysis, project management, etc. I feel that at the end of the workshop, there should be some space to think about what can be done with python when it is used in conjunction with all the other things that participants are learning. At the very beginning of the workshop we mention that python can be a way of "analyzing data, creating visualizations, composing interactive websites, scraping the internet, and engaging in distant reading of texts." Perhaps at the end of the workshop, we might point people towards other resources that helps them do these things?

**Comment by rafadavis** [comment]
- In the first Challenge, when explaining how to get out of python: In some windows machines, you need to Ctr-z + Enter to get out.

- In "Running Scripts": "To do this, we will begin to work with (a/the?) text editor"

- The transition between the python shell and writing scripts is very well explained!

- "The tools you've learned so far—the command line, Git, markdown, grep—are all designed to work well together, and the medium through which they all work is plain text." This kind of assumes they all did those workshops already? Maybe rephrase it with something like "the tools you will learn in this series of workshops"

- I think the random-book example is a bit unintuitive, and I have an idea that I think you're gonna like. I'll tell you on the phone because it is much easier than writing here. 

- This is nitty-picky, but: "Remember the proper syntax for creating a list includes square brakets with quotations and commas separating the list items." This phrasing may lead them to think that you always have to put items of a list with quotations. Maybe better to say "Remember the proper syntax for creating a list includes square brackets with commas separating the list items. Since the items are strings, they should also be inside quotes". Also, "brackets" is missing a "c". It is also misspelled in the section "doing things with lists"

- Nice library.py challenge!

- "This involves adding a while statement to your libary app". "R" missing in library

- I'm not sold on having that while loop challenge. The while loop was one of the things I wanted to cut from the original workshop. I think it is a bit unintuitive and not used too often.

If you do leave it, I think you need to teach how to interrupt (Ctrl-c), otherwise they will be stuck on an infinite loop. Also, teach them how to include a break option on the loop. 

There is another challenge later that you suggest the while loop. Same goes there.

- "The above output shows all the methods and attributes available to Python strings that can be accessed using the dot (.) syntax. When using dir()". Dir doesn't show all methods, shows some that they consider to be more important. But it does happen that you don't find the one you are looking for, so it is best to let them know of the possibility.

**Comment by gofilipa** [comment]
I've finished addressing all of the above, including adding a final challenge to get people started with scraping. 

I also implemented one of Rafa's suggestion to make the book loop a bit more intuitive. Rather than checking for the "correct" book, it checks for the field of study, then prints out a book. This is for the "Conditionals" lesson, though it also bleeds into the challenge in "Inputs." The code is below: 

```python
field = "Media Studies"
if field == "Media Studies":
    print("Grammophone, Film, Typewriter")
elif field == "Critical University Studies":
    print("The Undercommons")
elif field == "Textual Scholarship":
    print("Radiant Textuality")
elif field == "Critical Race Studies"
    print("The New Jim Code")
elif field == "DH Methodologies"
    print("Algorithmic Criticism")
else:
    print("I don't know what field you're talking about! I'm just a little program...")
```
